### PR TITLE
Add more wake lock tests

### PR DESCRIPTION
**Pull request by Honry**
- test in an insecure context
  - test for invoking 'cancel()' twice
  - test when document is hidden
  - test for applicability when screen is switched off

https://github.com/w3c/wake-lock/issues/111

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/20605)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/302262055)

File: wake-lock/wakelock-applicability-manual.https.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>wake lock applicability test</title>
+<link rel="help" href="https://w3c.github.io/wake-lock/#dfn-the-wake-lock-is-applicable">
+<meta name="flags" content="interact">
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<p>
+  Lock and turn off the screen, then turn it on and unlock the screen in a minute.
```
Comment (marcoscaceres) [inline wake-lock/wakelock-applicability-manual.https.html]: This is unclear, are you asking the tested to wait 60 seconds?

File: wake-lock/wakelock-applicability-manual.https.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>wake lock applicability test</title>
+<link rel="help" href="https://w3c.github.io/wake-lock/#dfn-the-wake-lock-is-applicable">
+<meta name="flags" content="interact">
+<script src=/resources/testharness.js></script>
```
Comment (marcoscaceres) [inline wake-lock/wakelock-applicability-manual.https.html]: nit: unquoted attributes.

File: wake-lock/wakelock-applicability-manual.https.html [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  await eventWatcher.wait_for("visibilitychange");
+  assert_true(document.hidden, "document is hidden when screen is locked");
+  assert_false(wakeLock.active, "the screen wake lock is not active when screen is switched off");
+
+  //unlock screen to fire 'visibilitychange'
+  await eventWatcher.wait_for("visibilitychange");
+  assert_false(document.hidden, "document is visiable when screen is unlocked");
+  assert_true(wakeLock.active, "the screen wake lock is active when screen is switched on again");
+  request.cancel();
+}, "The screen wake lock isn't applicable after the screen is manually swiched off"
```
Comment (marcoscaceres) [inline wake-lock/wakelock-applicability-manual.https.html]: Nit: "applicable" is a somewhat confusing term. Maybe "usable"? dunno.

File: wake-lock/wakelock-applicability-manual.https.html (hunk 1 above)
Comment (marcoscaceres) [reply wake-lock/wakelock-applicability-manual.https.html]: I theory, the tests will run by the time the tested turns the screen off and then on again. So likely they only have to wait maybe 1 or 2 seconds, right?

File: wake-lock/wakelock-cancel-twice.https.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+promise_test(async t => {
+  const wakeLock = await navigator.getWakeLock("screen");
+  const request = wakeLock.createRequest();
+  assert_true(wakeLock.active, "the activate is true when wake lock is acquired");
+  request.cancel();
+  assert_false(wakeLock.active, "the activate is false when wake lock is released");
+  //If the cancel() method has already been invoked on this object,
+  //abort these steps, no error fired
+  request.cancel();
+}, document.title);
```
Comment (marcoscaceres) [inline wake-lock/wakelock-cancel-twice.https.html]: Cute hack :) However, if we end up adding more tests to this file, people looking at the test might get annoyed if they have to scroll to the top to find out what the test does. Just inline the testable assertion here and maybe generalize the doc's title.

File: wake-lock/wakelock-insecure-context.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+<meta charset="utf-8">
+<title>Wake Lock API is not exposed in an insecure context</title>
+<link rel="help" href="https://w3c.github.io/wake-lock/#wake-locks">
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script>
+
+test(t => {
+  assert_false("getWakeLock" in navigator, "'getWakeLock' must not be exposed");
+}, document.title);
```
Comment (marcoscaceres) [inline wake-lock/wakelock-insecure-context.html]: as above.

**Review by marcoscaceres — APPROVED** [review]
Couple of nits, but these are looking great @Honry! 🏆 

I'd like to get a review from @andrey-logvinov before we merge. @andrey-logvinov, can you please double check that the algorithms are being fully tested?

File: wake-lock/wakelock-applicability-manual.https.html (hunk 1 above)
Comment (Honry) [reply wake-lock/wakelock-applicability-manual.https.html]: I didn't mean waiting, I just wanted to make sure the interaction should be complete within 60 seconds, before test timeout. Anyway, I will improve the description.

File: wake-lock/wakelock-applicability-manual.https.html (hunk 3 above)
Comment (Honry) [reply wake-lock/wakelock-applicability-manual.https.html]: Hmm.. This sentence is referenced from [spec](https://w3c.github.io/wake-lock/#dfn-the-wake-lock-is-applicable), I think we can leave it as is.

> The screen wake lock MUST NOT be applicable after the screen is manually swiched off by the user until it is switched on again. Manually switching off the screen MUST NOT affect the applicability of the system wake lock.

File: wake-lock/wakelock-cancel-twice.https.html (hunk 4 above)
Comment (Honry) [reply wake-lock/wakelock-cancel-twice.https.html]: :) Will fix.

File: wake-lock/wakelock-applicability-manual.https.html (hunk 3 above)
Comment (marcoscaceres) [reply wake-lock/wakelock-applicability-manual.https.html]: ah, thanks for clarifying... ok, yeah, it makes sense in the context of the spec.

**Comment by Honry** [comment]
> I'd like to get a review from @andrey-logvinov before we merge. @andrey-logvinov, can you please double check that the algorithms are being fully tested?

@andrey-logvinov, friendly ping. :)

**Comment by marcoscaceres** [comment]
Merging, as it's been quite a few days - @andrey-logvinov, could you please still have a look at this and file any bugs as appropriate?

**Pull request merged**